### ISSUE TITLE
Ability to display tooltips anchored to series positioning

### ISSUE THREAD

**Issue by dss010101** [highcharts/highcharts#17622]
# Description of the feature
With stockcharts, we often have multiple 'panes': i.e. one showing price, the others showing relative prices or indicators or studies.
To render these, we currently have to change the height and top of each series relative to each other and in context of the entire height.
Would be great if it were easier to be able to display the tooltip of each series relative to it's 'pane' - meaning relative to it's top.
Seems more intuitive to keep that information with the series itself.  I think

# Library related to the feature
To do this i think a couple of features/functionality would be nice for Highchart Stockchart
1. ability to anchor tooltips for that series relative to the series position on the chart
2. The functionality should take into consideration resizing of the chart/series when yAxis.resize.enabled is true.
3. Perhaps the concept of a "Pane" should be introduced.  A Pane would be a group of series that can be positioned together and where we can manage various aspects of those series together, including tooltips.  A Chart may have multiple panes, panes may contain multiple series.
4. A nice to have as well would be to be able to always have the tooltip, when anchored, visible and showing the last days (latest series point) values.

See attachment below for an example of this problem solved today - but with a lot of additional coding necessary and some hacks.

# Proof of Concept/Live example for the feature

----

**You can vote for this feature by adding a thumbs-up reaction to this post.**
![image](https://user-images.githubusercontent.com/855627/185448993-02d353bc-90e5-43dc-9647-a12371b14dd5.png)

**Comment by dss010101** [comment]
i wish it wasnt closed...any serious financial charting app should support this functionality easily

**Comment by hubertkozik** [comment]
Sure, let's reopen this ticket.

**Comment by dss010101** [comment]
I seem to be the only one interested in this, but i wanted to show that other financial charts all support this out of the box..to perhaps entice the feature approver Gods some incentives and motivations...

**tradingview**
![image](https://github.com/user-attachments/assets/bbbf7b31-58ef-4e31-a06f-f6d2ee9e3633)

**barchart.com**
![Screenshot 2024-12-17 004831](https://github.com/user-attachments/assets/c57c7f84-324c-4c85-8fdb-67d694e7e4d3)

stockchart.com
![Screenshot 2024-12-17 004657](https://github.com/user-attachments/assets/a274899f-5a40-4c00-a53e-2c1fac1bc41b)

**Comment by pawelfus** [comment]
Hello @dss010101 

Thank you for advocating for this feature! It’s already on our radar, and you can track the progress in this PR: [#22331](https://github.com/highcharts/highcharts/pull/22331).

In the meantime, you can achieve the output you mentioned using the [tooltip.positioner](https://api.highcharts.com/highstock/tooltip.positioner) function. For example, in this demo: [Stock Tools GUI](https://www.highcharts.com/demo/stock/stock-tools-gui), each series or technical indicator displays its tooltip in the top-left corner of its respective pane:

<img width="1073" alt="image" src="https://github.com/user-attachments/assets/240bd818-3399-4b5e-921d-6d194b244efc" />

Let us know if you need further guidance!

**Comment by TorsteinHonsi** [comment]
@dss010101 Some progress has been made here now, see https://jsfiddle.net/highcharts/pnvtc3mh/

**Comment by dss010101** [comment]
> @dss010101 Some progress has been made here now, see https://jsfiddle.net/highcharts/pnvtc3mh/

this is great news!  this will save me ALOT of code that has become difficult to maintain.
Can you tell me specifically what the changes is? i noticed one new attribute, 'relativeTo'.  is that all?
Also, it would be nice also if clicking on the series name would toggle the series much like the legend does (this also i am currently coding for in my hacked implementation).
Lastly, what source can i pull from to test this locally?

Thank you for taking this seriously!

**Comment by TorsteinHonsi** [comment]
My pleasure :)

> Can you tell me specifically what the changes are?

I added a new configuration object, [tooltip.position](https://github.com/highcharts/highcharts/pull/22331/files#diff-69aaab7f6160fdbb3c0ab85a4cf08b9d39566977be1a304d543ce712e0413d93R90-R97).

> Also, it would be nice also if clicking on the series name would toggle the series much like the legend does 

We will _probably_ not build in that functionality, but it should be possible to achieve it using the formatter. However, a side effect would be that once you hide the series, it is also hidden from the tooltip so you cannot show it again, unless you have a legend?
 
> Lastly, what source can i pull from to test this locally?

You'll find the URLs to github.highcharts.com under the HTML tab in the [fiddle](https://jsfiddle.net/highcharts/pnvtc3mh/). I should warn you it's still an early draft though, there's still some work to do.

**Comment by dss010101** [comment]
> We will probably not build in that functionality, but it should be possible to achieve it using the formatter. However, a side effect would be that once you hide the series, it is also hidden from the tooltip so you cannot show it again, unless you have a legend?

I actually have an implementation of this that works from when i recreated this functionality with custom code. i am able to toggle on/off a series.  i have to look at it to see how i implemented it. but here is a demo:

https://drive.google.com/file/d/1EZ5_TsJPtzct8m6kN8JGChmHC-aEw0o0/view?usp=sharing

i do wanted to also be able to highlight which series is bing hovered over much like the legend.  but wasn't able to.  there are some cases where there are many timeseries (i.e. full yield curve timeseries), where it becomes difficult to identify which series is what without the highlighting.

**Comment by TorsteinHonsi** [comment]
@dss010101 It would be interesting to see how this looks. I don't have access to the Google doc though. Can you set it up in jsFiddle?

**Comment by dss010101** [comment]
> @dss010101 It would be interesting to see how this looks. I don't have access to the Google doc though. Can you set it up in jsFiddle?

try viewing it now.  its lot of custom code. i will try to setup something..i may have in prior questions. not sure..

**Comment by TorsteinHonsi** [comment]
@dss010101 I can see it now 👍 . I understand how it works. So basically it behaves like the legend, so when clicked it hides the series and the series stays in the "tooltip".

**Comment by dss010101** [comment]
> @dss010101 I can see it now 👍 . I understand how it works. So basically it behaves like the legend, so when clicked it hides the series and the series stays in the "tooltip".

right. only thing it doesnt do right now like the tool tip is highlight which series is being hovered over...for example if you are hovering over the actual series, it should bolden or do some other form of highlighting of the tool tip and vice versa if you are hovering over one of the series tooltip.
i feel that can be done, but i havent had time to try and implement it.

**Comment by pawelfus** [comment]
You can try out [`chart.hoverSeries`](https://api.highcharts.com/class-reference/Highcharts.Chart#hoverSeries) property. For example, using `pointFormatter` it could be something like this: https://jsfiddle.net/BlackLabel/4dc9h8jL/ Snippet:

```ts
      tooltip: {
        pointFormatter: function () {
          const bold = this.series.chart.hoverSeries === this.series;

          return `
            ${bold ? "<b>" : ""}
              <span style="color: ${this.series.color}">${this.series.name}</span>: ${this.y} (${this.change.toFixed(2)}%)
            ${bold ? "</b>" : ""}`
        },
        valueDecimals: 2,
        split: true,
      },
```

And from tooltip, you can call [`series.setState('hover')`](https://api.highcharts.com/class-reference/Highcharts.Series#setState) for series you want to highlight.

**Comment by dss010101** [comment]
> You can try out [`chart.hoverSeries`](https://api.highcharts.com/class-reference/Highcharts.Chart#hoverSeries) property. For example, using `pointFormatter` it could be something like this: https://jsfiddle.net/BlackLabel/4dc9h8jL/ Snippet:
> 
> ```ts
>       tooltip: {
>         pointFormatter: function () {
>           const bold = this.series.chart.hoverSeries === this.series;
> 
>           return `
>             ${bold ? "<b>" : ""}
>               <span style="color: ${this.series.color}">${this.series.name}</span>: ${this.y} (${this.change.toFixed(2)}%)
>             ${bold ? "</b>" : ""}`
>         },
>         split: true,
>       },
> ```
> 
> And from tooltip, you can call [`series.setState('hover')`](https://api.highcharts.com/class-reference/Highcharts.Series#setState) for series you want to highlight.

interesting. i will give it a shot. thank you

**Comment by pawelfus** [comment]
https://github.com/highcharts/highcharts/pull/22331 has been merged and is available since v12.2.0

Now it is possible, to simply use `tooltip.split` with `tooltip.position.relativeTo` to configure the desired design, demos:
- https://jsfiddle.net/BlackLabel/o4w1nk9L/
- https://www.highcharts.com/demo/stock/stock-tools-gui

More config options are available, check out the API ref: https://api.highcharts.com/highcharts/tooltip.position

Let us know if you find any issues in the current implementation!
